src/main/cpp/checkpoint/CheckPoint.h

```diff
@@ -0,0 +1,96 @@
+/*
+ * CheckPoint.h
+ *
+ *  Created on: Mar 26, 2017
+ *      Author: cedric
+ */
+
+#ifndef CHECKPOINT_H_INCLUDED
+#define CHECKPOINT_H_INCLUDED
+
+#include <memory>
+#include <vector>
+#include "calendar/Calendar.h"
+#include "core/Cluster.h"
+#include <hdf5.h>
+#include "pop/Population.h"
+#include "sim/SimulationConfig.h"
+
+namespace stride {
+namespace checkpoint {
+
+class CheckPoint
+{
+public:
+	/// Creates the wanted file and immediately closes it. It will overwrite a file if one of the same name already
+	/// exists.
+	void CreateFile();
+
+	/// Constructor The string is the file for the checkpoints. The unsigned int is the interval between
+	/// checkpoints. An interval of 0 makes no checkpoints.
+	CheckPoint(const std::string&, unsigned int interval = 5);
+
+	/// Loads a checkpoint from the file in the constructor. The unsigned int tells which checkpoint to use.
+	Population LoadCheckPoint(unsigned int, std::vector<std::vector<Cluster>>&);
```

Comment:
yes I am editing the list of a list of clusters.
The inner list will be clusters of the same clustertype
The outer list is a list of lists with different clustertypes

I made it a list of a list to make it a little more generic to allow new clustertypes.